.github/workflows/frontend-applications/release-generic.yml

```diff
@@ -0,0 +1,147 @@
+name: Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      releaseVersion:
+        description: 'Release version (format: vX.X)'
+        required: true
+        type: string
+      commitSha:
+        description: 'SHA of the commit to release from'
+        required: true
+        type: string
+      dockerImage:
+        description: 'Name of the Docker image (e.g., organization/app-name)'
+        required: true
+        type: string
+      dockerUsername:
+        required: true
+        type: string
+      githubappId:
+        description: 'GitHub App ID for authentication'
+        required: true
+        type: string
+    secrets:
+      githubappPrivateKey:
+        required: true
+      docker-token:
+        required: true
+      sonar-token:
+        required: true
+
+env:
+  NODE_OPTIONS: '--max_old_space_size=4096'
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Generate GitHub App Token
+        id: app-token
+        uses: actions/create-github-app-token@v1
+        with:
+          app-id: ${{ inputs.githubappId }}
+          private-key: ${{ secrets.githubappPrivateKey }}
+
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ steps.app-token.outputs.token }}
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18.x'
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Extract Version Numbers
+        run: |
+          if ! [[ ${{ inputs.releaseVersion }} =~ ^v([0-9]+)\.([0-9]+)$ ]]; then
+            echo "Error: Invalid version format. Must be vX.X (e.g., v1.2)"
+            exit 1
+          fi
+          
+          MAJOR="${BASH_REMATCH[1]}"
+          MINOR="${BASH_REMATCH[2]}"
+          
+          echo "RELEASE_MAJOR_VERSION=${MAJOR}" >> $GITHUB_ENV
+          echo "RELEASE_MINOR_VERSION=${MINOR}" >> $GITHUB_ENV
+          echo "RELEASE_VERSION=${MAJOR}.${MINOR}.0" >> $GITHUB_ENV
+          echo "RELEASE_BRANCH=release-v${MAJOR}.${MINOR}" >> $GITHUB_ENV
+
+      - name: Create Release Branch
+        run: |
+          git checkout -b ${{ env.RELEASE_BRANCH }} ${{ inputs.commitSha }}
+
+      - name: Update Version and Tag
+        run: |
+          # Configure git for commits
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          
+          # Use npm version to update version, create commit and tag
+          npm version ${{ env.RELEASE_VERSION }}
+          
+          # Push both the tag and the branch
+          git push origin v${{ env.RELEASE_VERSION }}
+          git push origin ${{ env.RELEASE_BRANCH }}
```

Comment:
You should push only after all checks and sonar and docker image are done, as it is in backend-applications